SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityDeployChecker.java

```diff
@@ -611,6 +611,11 @@ private SingularityDeployResult checkDeployProgress(final SingularityRequest req
       return checkCanMoveToNextDeployStep(request, deploy, pendingDeploy, updatePendingDeployRequest);
     }
 
+    final DeployHealth deployHealth = deployHealthHelper.getDeployHealth(request, deploy, deployActiveTasks, true);
+    if (request.getRequestType().equals(RequestType.WORKER) && deployHealth.equals(DeployHealth.UNHEALTHY)) {
```

Comment:
Can do == for enum comparison

Reply:
Other thing, can we leave a note somehow, or maybe a mini method with a descriptive name so we know this line is here specifically to catch non-web health check failures?